crates/cheatcodes/src/test/expect.rs

```diff
@@ -434,11 +434,16 @@ pub(crate) fn handle_expect_emit(state: &mut Cheatcodes, log: &alloy_primitives:
     let expected_topic_0 = expected.topics().first();
     let log_topic_0 = log.topics().first();
 
-    if expected_topic_0
-        .zip(log_topic_0)
-        .map_or(false, |(a, b)| a == b && expected.topics().len() == log.topics().len())
-    {
-        // Match topics
+    if expected_topic_0.zip(log_topic_0).map_or(
+        // Anonymous events are a special case where topic 0 is not generated.
+        // If the anonymous event has no other topics we will still attempt to match the source
+        // address and data if requested.
+        expected_topic_0.is_none() && log_topic_0.is_none(),
+        // If the topics match and the length of the topic list is the same, we can proceed to
+        // check.
+        |(a, b)| a == b && expected.topics().len() == log.topics().len(),
+    ) {
```

Comment:
I see, this makes sense to me now,

I feel, the map or is a bit hard to read here, I'd like to replace this with a match statement instead